flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputSelectableProcessor.java

```diff
@@ -286,64 +316,33 @@ private void checkAndSetAvailable(int inputIndex) {
 		}
 	}
 
-	/**
-	 * @return false if both of the inputs are finished, true otherwise.
-	 */
-	private boolean waitForAvailableInput(InputSelection inputSelection)
-		throws ExecutionException, InterruptedException, IOException {
-
+	private CompletableFuture<?> waitForAvailableInput(InputSelection inputSelection) {
 		if (inputSelection.isALLMaskOf2()) {
 			return waitForAvailableEitherInput();
 		} else {
-			waitForOneInput(
+			return waitForOneInput(
 				(inputSelection.getInputMask() == InputSelection.FIRST.getInputMask()) ? input1 : input2);
-			return true;
 		}
 	}
 
-	private boolean waitForAvailableEitherInput()
-		throws ExecutionException, InterruptedException {
-
-		CompletableFuture<?> future1 = input1.isFinished() ? UNAVAILABLE : input1.isAvailable();
-		CompletableFuture<?> future2 = input2.isFinished() ? UNAVAILABLE : input2.isAvailable();
-
-		if (future1 == UNAVAILABLE && future2 == UNAVAILABLE) {
-			return false;
+	private CompletableFuture<?> waitForAvailableEitherInput() {
+		if (input1.isFinished() && input2.isFinished()) {
+			return AVAILABLE;
```

Comment:
I think this should not happen, but if it's just to cover all cases, I'd prefer to return here `UNAVAILABLE`.

Additional note about renaming this method: "either" means **mutually exclusive** (out of two options) in English. I'd name this method something like `getAnyInputAvailableFuture()`.